### PR TITLE
[16.0][MIG] l10n_fr_department and l10n_fr_department_oversea

### PR DESCRIPTION
**Pull request by alexis-via**
fix bug https://github.com/OCA/l10n-france/issues/384 : field department_id on res.partner has been renamed to country_department_id

### CODE REVIEW

**Comment by alexis-via** [comment]
Requires https://github.com/OCA/l10n-france/pull/398

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Thanks for porting this module. 

Migration script is missing.

Otherwise, LGTM.

**Comment by alexis-via** [comment]
As it is a computed field, a migration script is not required, but it could save same time during migration.

File: l10n_fr_department/model/res_partner.py [hunk 1]
```diff
@@ -10,16 +10,19 @@
 class ResPartner(models.Model):
     _inherit = "res.partner"
 
-    department_id = fields.Many2one(
+    # In v16, this field was renamed from department_id to country_department_id
+    # to avoid a field name conflict with the module partner_contact_department
+    # from https://github.com/OCA/partner-contact
```
Comment (ivantodorovich) [inline l10n_fr_department/model/res_partner.py]: These comments belong to the commit message or to the PR itself.

They're only useful when reviewing/analyzing changes from previous code.
But they have no use at all in the present/future code

File: l10n_fr_department/view/res_country_department.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <!--
-  Copyright 2013-2020 GRAP (http://www.grap.coop)
+  Copyright 2013-2022 GRAP (http://www.grap.coop)
```
Comment (ivantodorovich) [inline l10n_fr_department/view/res_country_department.xml:3]: FTR you could save yourself from ever having to update these Copyright lines.
When reading the copyright comment, the dates are to be read as "starting.." (and legally it'll be valid for up to 70 years from that point)

eg: `Copyright 2013 GRAP (http://www.grap.coop)`

Means it's copyright is valid from 2013 to 2083.
There's no need at all to put an end date, that end up being just noisy diffs, and it's totally useless.


https://stackoverflow.com/questions/2390230/do-copyright-dates-need-to-be-updated

**Comment by legalsylvain** [comment]
> As it is a computed field, a migration script is not required, but it could save same time during migration.

Indeed. But a little weird to raise the recompute of an existing field. And on res.partner table, it can take time. Better to use rename_fields of openupgradelib.

**Comment by alexis-via** [comment]
I added and tested the migration script. Requires a recent version of openupgradelib to support the new system for translations.

**Review by legalsylvain — APPROVED** [review]
Thanks !

**Pull request merged**